integration/sawtooth_integration/tests/test_poet_smoke.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import unittest
+import subprocess
+import shlex
+import time
+import logging
+import operator # used by verifier
+# -- used by rest_api callers --
+import urllib.request
+import urllib.error
+import json
+from base64 import b64decode
+
+import cbor
+
+from sawtooth_integration.intkey.intkey_message_factory \
+    import IntkeyMessageFactory
+from sawtooth_cli.rest_client import RestClient
+
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.DEBUG)
+
+# this needs to be coordinated with the test's yaml file
+VALIDATOR_COUNT = 3
+
+INITIAL_WAIT_TIME = 20
+TIME_BETWEEN_BATCHES = 1
+CATCH_UP_TIME = 20
+
+# if this is too high, the test will time out
+BATCH_COUNT = 20
+
+class TestPoetSmoke(unittest.TestCase):
+    def setUp(self):
+        urls = ['http://rest-api-{}:8080'.format(i)
+                for i in range(VALIDATOR_COUNT)]
+
+        self.clients = [IntkeyClient(url) for url in urls]
+
+    def test_poet_smoke(self):
+        '''
+        $VALIDATOR_COUNT validators are started, each with config,
+        intkey, and validator registry transaction processors. After
+        waiting for the validators to register, do the following:
+
+        1) Send a batch of intkey 'set' transactions to one validator.
+
+        2) Send one batch of intkey 'inc' transactions to each validator
+           one after the other $BATCH_COUNT times.
+
+        3) Loop through the validators, sending each one $BATCH_COUNT
+           batches of intkey 'inc' transactions.
+
+        4) Assert that the validators are in consensus with each.
+        '''
+        populate, increment = _make_txns()
+
+        # wait for validators to register
+        time.sleep(INITIAL_WAIT_TIME)
```

Comment:
It needs to wait for all the validators to get the genesis block, since a validator can't be queried for its blocks if it doesn't have any. I don't know if there's anything better to do than to wait for them.